src/installer/managed/Microsoft.NET.HostModel/Bundle/TargetInfo.cs

```diff
@@ -53,6 +54,12 @@ public bool IsNativeBinary(string filePath)
             return IsLinux ? ElfUtils.IsElfImage(filePath) : IsOSX ? MachOUtils.IsMachOImage(filePath) : PEUtils.IsPEImage(filePath);
         }
 
+        public string GetAppName(string hostName)
```

Comment:
I think we should change this to return just the app name - so without the `.dll`. After all that's what the caller wants.
The assert is I think partially wrong - it can in theory happen that the file is not called `.exe` on windows, or that it will be called `.exe` on Linux - I can still make it run (I think - does CreateProcess block running something which is not called .exe?).

I actually think we should make this a runtime check and fail if it's not the case since we won't know what to do with it otherwise.

Reply:
OK I'll change the method to return assemblyname without the `.dll`.
Once the SDK is updated to pass the AssemblyName, this method  can be removed (and maybe moved to the `BundleHelper` test infrastructure. So, I won't add an exception for this check.